uavcan/src/transport/can/mod.rs

```diff
@@ -162,9 +162,7 @@ impl<'a, C: embedded_time::Clock> CanIter<'a, C> {
                     return Err(TxError::AnonNotSingleFrame);
                 }
 
-                CanMessageId::new(transfer.priority, transfer.port_id, node_id)
-                    .to_u32()
-                    .unwrap()
+                CanMessageId::new(transfer.priority, transfer.port_id, node_id).0
```

Comment:
Convert to plain type later. First assign as Message-ID type. For convenience add a function to convert to native type.

Reply:
I did this so all the match arms were the same. I don't see an easy way to do the typing other than adding an enum on top of these types but it doesn't seem worth it just for this instance.